### ISSUE TITLE
Regression in `2.4.12` binary parser

### ISSUE THREAD

**Issue by ritch** [primus/primus#368]
Sorry for the lack of detail... but I'm fairly certain there is a regression in a patch release around 2.4.12 or so. I'm getting a lot of weird behavior and errors after upgrading. When I downgrade all seems to work.

My basic setup

```js
  var primus = new Primus(httpServer, {
    transformer: 'engine.io',
    parser: 'binary'
  });
```

 - no data event on the server
 - (or) an exception when trying to decode the incoming message (cannot call undefined)

I'll update this issue when I get some time.

**Comment by ritch** [comment]
Actually on second glance this is more likely a fix to a dependency than a regression introduced in primus.

**Comment by lpinca** [comment]
@ritch it could be the `binary-pack` module. Make sure that you are using version `0.0.3`. I've published `binary-pack@1.0.0` but it will work in the next version of Primus, not the current one.

**Comment by ritch** [comment]
@lpinca thanks for the quick response... that was it. Lets close this.

**Comment by lpinca** [comment]
@ritch I've published `binary-pack@1.0.1` which is compatible with older versions of primus.
It is also possible to use older versions of `binary-pack` with primus master.

Thank you for your report.
